bookkeeper-server/src/test/java/org/apache/bookkeeper/client/CorrectEnsemblePlacementCmdTest.java

```diff
@@ -0,0 +1,235 @@
+    private BookKeeper.DigestType digestType = BookKeeper.DigestType.CRC32;
+    private static final String PASSWORD = "testPasswd";
+
+    public CorrectEnsemblePlacementCmdTest() throws Exception {
+        super(0);
+        baseConf.setLedgerStorageClass(DbLedgerStorage.class.getName());
+        baseConf.setGcWaitTime(60000);
+        baseConf.setFlushInterval(1);
+    }
+
+    /**
+     * list of entry logger files that contains given ledgerId.
+     */
+    @Test
+    public void testArgument() throws Exception {
+        startNewBookie();
+
+        @Cleanup
+        final BookKeeper bk = new BookKeeper(baseClientConf, zkc);
+        @Cleanup
+        final LedgerHandle lh = createLedgerWithEntries(bk, 10, 1, 1);
+
+        final String[] argv1 = { "correct-ensemble-placement", "--ledgerids", String.valueOf(lh.getId()),
+                "--skipOpenLedgers", "--force"};
+        final String[] argv2 = { "correct-ensemble-placement", "--ledgerids", String.valueOf(lh.getId()),
+                "--skipOpenLedgers", "--force", "--dryrun"};
+        final BookieShell bkShell =
+                new BookieShell(LedgerIdFormatter.LONG_LEDGERID_FORMATTER, EntryFormatter.STRING_FORMATTER);
+        bkShell.setConf(baseClientConf);
+
+        assertEquals("Failed to return exit code!", 0, bkShell.run(argv1));
+        assertEquals("Failed to return exit code!", 0, bkShell.run(argv2));
+    }
+
+    @Test
+    public void testCorrectEnsemblePlacementByRackaware() throws Exception {
+        final int ensembleSize = 7;
+        final int quorumSize = 2;
+
+        final BookieSocketAddress addr1 = new BookieSocketAddress("127.0.0.1", 3181);
+        final BookieSocketAddress addr2 = new BookieSocketAddress("127.0.0.2", 3181);
+        final BookieSocketAddress addr3 = new BookieSocketAddress("127.0.0.3", 3181);
+        final BookieSocketAddress addr4 = new BookieSocketAddress("127.0.0.4", 3181);
+        final BookieSocketAddress addr5 = new BookieSocketAddress("127.0.0.5", 3181);
+        final BookieSocketAddress addr6 = new BookieSocketAddress("127.0.0.6", 3181);
+        final BookieSocketAddress addr7 = new BookieSocketAddress("127.0.0.7", 3181);
+        final BookieSocketAddress addr8 = new BookieSocketAddress("127.0.0.8", 3181);
+        final BookieSocketAddress addr9 = new BookieSocketAddress("127.0.0.9", 3181);
+
+        final Set<BookieId> writableBookies = new HashSet<>();
+        writableBookies.add(addr1.toBookieId());
+        writableBookies.add(addr2.toBookieId());
+        writableBookies.add(addr3.toBookieId());
+        writableBookies.add(addr4.toBookieId());
+        writableBookies.add(addr5.toBookieId());
+        writableBookies.add(addr6.toBookieId());
+        writableBookies.add(addr7.toBookieId());
+        writableBookies.add(addr8.toBookieId());
+        writableBookies.add(addr9.toBookieId());
+
+        // add bookie node to resolver
+        StaticDNSResolver.reset();
```

Comment:
we should do this also in a "After" method, in order to ensure that we do not pollute the test environment